### PR TITLE
upgrade gunicorn and gevent & fix flake8

### PR DESCRIPTION
**Pull request by Xiangs18**
Update these two dependencies (gunicorn/gevent) so that `make start` can run successfully.

### CODE REVIEW

File: feeds/external_api/groups.py [hunk 1]
```diff
@@ -50,8 +50,8 @@ def validate_group_id(group_id: str) -> bool:
         return res['exists']
     elif 'error' in res:
         raise GroupsError(
-            "Error while looking up group id: " +
-            res['error'].get('message', 'no message available')
+            f"Error while looking up group id: "
+            f"{res['error'].get('message', 'no message available')}"
```
[the remark this answers is not in the document]
Comment (Xiangs18) [reply feeds/external_api/groups.py:54]: How would you do differently to fix flake8 complain here?

[the remark this answers is not in the document]
Comment (MrCreosote) [reply feeds/external_api/groups.py:54]: Why is flake 8 complaining?

**Comment by Xiangs18** [comment]
> What prevented it from starting before? It must've started since presumably we're running with those versions in CI / prod

If we deploy docker pull ghcr.io/kbase/feeds:develop to CI, the server will fail. This PR aims to fix the issue. You may see a message similar to the following in the logs:

Screenshot 2024-11-18 at 6 26 23 PM

**Comment by MrCreosote** [comment]
So how is the current server running?

File: feeds/external_api/groups.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Xiangs18) [reply feeds/external_api/groups.py:54]: feeds/external_api/groups.py:53:49: W504 line break after binary operator

[the remark this answers is not in the document]
Comment (MrCreosote) [reply feeds/external_api/groups.py:54]: put the + sign on the next lne

**Comment by Xiangs18** [comment]
My understanding is that the current server runs on a Python version < 3.8, which is compatible with very old versions of Gunicorn and Gevent.

File: feeds/external_api/groups.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Xiangs18) [reply feeds/external_api/groups.py:54]: that's exactly what did, then you got another complain

[the remark this answers is not in the document]
Comment (MrCreosote) [reply feeds/external_api/groups.py:54]: What's the complaint?

**Comment by MrCreosote** [comment]
Ok so upgrading python broke things? Is is necessary to upgrade python?

File: feeds/external_api/groups.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Xiangs18) [reply feeds/external_api/groups.py:54]: feeds/external_api/groups.py:54:13: W503 line break before binary operator

**Comment by Xiangs18** [comment]
> Ok so upgrading python broke things? Is is necessary to upgrade python?

Yes, updating Python broke things because old dependencies are no longer compatible. However, it's good hygiene to keep Python up to date—at least no more Python 3.6. Additionally, other python repositories such as `handle_service2` and `idmappingservice` have made the update and now sit beautifully in CI.

File: feeds/external_api/groups.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MrCreosote) [reply feeds/external_api/groups.py:54]: So this is why I don't typically use linters, they're too stupid. Add an ignore to that line.

**Comment by MrCreosote** [comment]
Fair enough

File: feeds/external_api/groups.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Xiangs18) [reply feeds/external_api/groups.py]: 👍

**Review by MrCreosote — COMMENTED** [review]
LGTM but @briehl should review

File: requirements.txt [hunk 2]
```diff
@@ -1,8 +1,8 @@
 flask==2.0.3
 werkzeug==2.0.3
 requests==2.20.0
-gunicorn==19.9.0
-gevent==1.3.7
+gunicorn==22.0.0
+gevent==24.2.1
```
Comment (briehl) [inline requirements.txt:5]: That's an update!

**Review by briehl — APPROVED** [review]
LGTM

**Pull request merged**